modules/Installer/AWS.Tools.Installer.psm1

```diff
@@ -228,174 +245,117 @@ function Get-AvailableModuleVersion {
+            $manifestFileStream = [System.IO.File]::OpenWrite($temporaryManifestFilePath)
+            $entryStream.CopyTo($manifestFileStream)
+            $manifestFileStream.Close();
+
+            if ($PSVersionTable.PSVersion.Major -lt 6 -or $IsWindows) {
+                [System.Management.Automation.Signature]$manifestSignature = Get-AuthenticodeSignature -FilePath $temporaryManifestFilePath
+                if ($manifestSignature.Status -eq 'Valid' -and $manifestSignature.SignerCertificate.Subject -eq $script:AWSToolsSignatureSubject) {
+                    Write-Verbose "[$($MyInvocation.MyCommand)] Manifest signature correctly validated"
                 }
                 else {
-                    Write-Verbose "[$($MyInvocation.MyCommand)] Authenticode signature can only be vefied on Windows, skipping"
+                    throw "Error validating manifest signature for $($Name)"
                 }
+            }
+            else {
+                Write-Verbose "[$($MyInvocation.MyCommand)] Authenticode signature can only be vefied on Windows, skipping"
+            }
 
-                $manifestData = Import-PowerShellDataFile $temporaryManifestFilePath
-
-                if ($manifestData.PrivateData.MinAWSToolsInstallerVersion) {
-                    $minVersion = Get-CleanVersion $manifestData.PrivateData.MinAWSToolsInstallerVersion
-                    if ($minVersion -gt $script:CurrentMinAWSToolsInstallerVersion) {
-                        throw "$($moduleInfo.Name) version $RequiredVersion requires at least AWS.Tools.Installer version $minVersion. Run 'Update-Module AWS.Tools.Installer -Force'."
-                    }
-                }
+            [PSObject]$manifestData = Import-PowerShellDataFile $temporaryManifestFilePath
 
-                $saveAWSToolsModuleParams = @{
-                    RequiredVersion        = $RequiredVersion
-                    TemporaryRepoDirectory = $TemporaryRepoDirectory
-                    SavedModules           = $SavedModules
-                }
-                $manifestData.RequiredModules | ForEach-Object {
-                    Write-Verbose "[$($MyInvocation.MyCommand)] Found dependency $($_.ModuleName)"
-                    Save-AWSToolsModule -Name $_.ModuleName @saveAWSToolsModuleParams @proxyParams | Out-Null
+            if ($manifestData.PrivateData.ContainsKey('MinAWSToolsInstallerVersion')) {
+                [System.Version]$minVersion = Get-CleanVersion $manifestData.PrivateData.MinAWSToolsInstallerVersion
+                if ($minVersion -gt $script:CurrentMinAWSToolsInstallerVersion) {
+                    throw "$($Name) version $RequiredVersion requires at least AWS.Tools.Installer version $minVersion. Run 'Update-Module AWS.Tools.Installer -Force'."
                 }
+            }
 
-                #Returning the property capitalized module name.
-                $moduleInfo.Name
+            $manifestData.RequiredModules | ForEach-Object {
```

Comment:
I can't get this to fail, `RequiredModules` is always expected to be present (even if it would be an empty list for `AWS.Tools.Common`. I will avoid changing this for now, I don't want to risk masking some other issue. Please attach a `-Verbose` log if you can still reproduce it.